supervisor/plugins/base.py

```diff
@@ -160,6 +165,8 @@ async def load(self) -> None:
             await self.instance.attach(
```

Comment:
This is attaching to a running container. It seems `DockerError` is raised if that fails. So, when we adopt a new disk, I expect this to raise `DockerError`. This in turn calls `install()` :thinking: 

How is `check_image` getting called here when we actually need a new image? :thinking:

Reply:
Hm, in the add-on case, we suppress `DockerError`, so I see how this works in that case. But in general, it seems to me we want to check the image before attach no? Or maybe as part of attach?

Reply:
That attaches to a container if it exists. If it does not exist then it just looks for the expected image and stores its metadata instead. It only raises an exception if both the container and the expected image do not exist